### ISSUE TITLE
Feature Request: Average pilots in space in the last 30 minutes

### ISSUE THREAD

**Issue by ezi0o** [ccpgames/esi-issues#391]
### Request

Since forever there are two stats that are **only available in-game in the in-game-map** and never included on the api/xlm for external use (dotlan for example)

- Average pilots in space in the last 30 minutes
- Cynosural fields

could it finally be made available in the esi ? i mean if it has been available in the game forever why not make it accessible outgame? Was this intended or just an omission ?

thanks guys

**Comment by Risingson** [comment]
There are more than those 2:

+ 24 hour stats for NPCs killed/Player Kills/Pods/Jumps
+ Number of Pilots currently docked and active

Also not available oog are the Development Indices.

**Comment by Blacksmoke16** [comment]
Industry cost indices are here:  https://esi.tech.ccp.is/latest/#!/Industry/get_industry_systems

**Comment by Risingson** [comment]
![image](https://cloud.githubusercontent.com/assets/5108701/26037609/223dd4e4-38f7-11e7-837d-5c7f4dd431d7.png)

**Comment by Blacksmoke16** [comment]
Ah fair enough.  You can currently get the Defence Multiplier but the each individual index.

**Comment by gitAskur** [comment]
Since ESI (and CREST but let's not talk about CREST) actually communicates with the running simulation, giving simulation data like this is finally possible. The technical nature of the XML API was such that it was simply impossible to do (nothing is impossible, juts impractical), and that is no longer the case.

That being said, there´s most likely a solid reason for why that which we will not talk about did not provide that in-game functionaltiy and that reason is that a whole lot of stuff relies on certain things that only make sense in a eve-client<->eve-cluster environment. So I wouldn't wanna say it's easy to get all the map statistics in there - but we can get them in there.

I'll assign this to me and take a look once I'm done with a few other things. If this is still assigned to me on Monday, and this is the last comment from me, feel free to notify me and start the workers revolution.

**Comment by gitAskur** [comment]
I did not find time to do this today. Feel free to notify me on Monday, and start the workers revolution.

**Comment by ezi0o** [comment]
*bump*

Again thanks for taking to time to check this out, map stats in ESI would rock

**Comment by ezi0o** [comment]
*bump*

Again thanks for taking to time to check this out, map stats in ESI would
rock

**Comment by skyride** [comment]
I'm strongly against this being in the API. Poked a couple friends on the CSM about it, there's some solid game design reasons for it but it sounds like you're already checking it out. 👍

**Comment by gitAskur** [comment]
I am looking at this solely from the tech perspective.

There is an omnipresent "Game Design must approve this" caveat to anything we talk about here, and so far that has not changed. I just like to be able to tell people if something is possible to do when they start thinking about whether we should do it.

**Comment by ezi0o** [comment]
*bump*
any game design team feedback yet?
thx

**Comment by gitAskur** [comment]
Yes... yes there is. I requested clarification on it but from what I gathered this can happen in some delayed form.

**Comment by ezi0o** [comment]
great news!
maps stats such as NPC kills, jumps & all are i think refreshed from the api every hour.
people in space, every 30 minutes (says the title)

**Comment by gitAskur** [comment]
Got the clarification back and GD is happy to provide this out of client on a 24 hour delay to avoid facilitating automatic detection of things.

**Comment by Robbilie** [comment]
so 30m cache time and 24h delay? :D

**Comment by skyride** [comment]
Yeah surely it'd be better to just aggregate it over a longer period of time then? Like the 24hr NPC Kills data.

**Comment by Robbilie** [comment]
GET /universe/system_kills/ is cached for 1h

**Comment by ezi0o** [comment]
ok so this 24h delay concerns only the "player in space in the last 30 minutes" or all map stats (npc kills, jumps etc..) ?
because if it is, it's a MAJOR regression compared to was has been in the api for the past 10+ years

**Comment by gitAskur** [comment]
I´ll see if I can get the GD people to come over here and have a discussion about this.

**Comment by gitAskur** [comment]
Functional Parity will be maintained for pre-existing calls.

**Comment by ezi0o** [comment]
I've noticed that the number of NPC kills (updated every hour in the api for the past 10 years, and a major content generator for all pvp groups) is not present in the new ESI, would it be possible to add ?

additionnaly, would it be possible, as we already have jumps/kills/npc stats, to add a new one which would be "m2 mined" ?

**Comment by ccp-zoetrope** [comment]
Looking back through the GD discussion about this the request for average pilots in space in the past 30 minutes was not accepted. It was OK'd to have cynos in the last 24 hours.

**Comment by skyride** [comment]
Sounds good to me. Interesting/useful statistic but no where near game breaking.

**Comment by Risingson** [comment]
When doing cynos in the last 24 hours, then maybe adding kills/pods/npcs/jumps for last 24 hours would be cool alongside (client side apps without server-backend cannot aggregate this).

**Comment by gordonzero** [comment]
> It was OK'd to have cynos in the last 24 hours.

So eventually for null sec this would tell people what systems are cyno jammed?

If you do aggregate this data over a 24 hr period it needs to be in XXX cynos lit in the past 24 hrs.

Just my 2 cents on this topic.

**Comment by Seavert** [comment]
Is it possible to show average pilots in space in the past **60** minutes? Worse than in game, but consistent with `_kills` and `_jumps` endpoints :)

**Comment by lexa0303** [comment]
Any updates on this ? :)
